kornia/feature/laf.py

```diff
@@ -0,0 +1,381 @@
+        >>> input = torch.ones(1, 5, 2, 3)  # BxNx2x3
+        >>> output = kornia.get_laf_scale(input)  # BxNx1x1
+    """
+    n_dims = len(A.size())
+    eps = 1e-10
+    if (n_dims != 4):
+        raise TypeError(
+            "LAF shape should be must be [BxNx2x3]. "
+            "Got {}".format(A.size())
+        )
+    out = A[:, :, 0:1, 0:1] * A[:, :, 1:2, 1:2] -\
+          A[:, :, 1:2, 0:1] * A[:, :, 0:1, 1:2] + eps  # noqa: E127
+    return out.abs().sqrt()
+
+
+def make_upright(A: torch.Tensor) -> torch.Tensor: # noqa: E128
+    """
+    Rectifies the affine matrix, so that it becomes upright
+    Args:
+        A: (torch.Tensor): tensor of LAFs.
+
+    Returns:
+        torch.Tensor: tensor of same shape.
+
+    Shape:
+        - Input: :math:`(B, N, 2, 3)`
+        - Output:  :math:`(B, N, 2, 3)`
+    Example:
+        >>> input = torch.ones(1, 5, 2, 3)  # BxNx2x3
+        >>> output = kornia.make_upright(input)  #  BxNx2x3
+    """
+    n_dims = len(A.size())
+    if (n_dims != 4):
+        raise TypeError(
+            "LAF shape should be must be [BxNx2x3]. "
+            "Got {}".format(A.size())
+        )
+    det = get_laf_scale(A)
+    eps = 1e-10
+    b2a2 = torch.sqrt(A[:, :, 0:1, 1:2]**2 + A[:, :, 0:1, 0:1]**2)
+    A1_ell = torch.cat([(b2a2 / det).contiguous(),  # type: ignore
+                        torch.zeros_like(det)], dim=3)  # type: ignore
+    A2_ell = torch.cat([((A[:, :, 1:2, 1:2] * A[:, :, 0:1, 1:2] +
+                          A[:, :, 1:2, 0:1] * A[:, :, 0:1, 0:1]) / (b2a2 * det)),
+                          (det / b2a2).contiguous()], dim=3)  # type: ignore
+    return torch.cat([torch.cat([A1_ell, A2_ell], dim=2),
+                      A[:, :, :, 2:3]], dim=3)
+
+
+def ell2LAF(ells: torch.Tensor) -> torch.Tensor:
+    """
+    Converts ellipse regions to LAF format
+    Args:
+        A: (torch.Tensor): tensor of ellipses in Oxford format [x y a b c].
+
+    Returns:
+        LAF: (torch.Tensor) tensor of ellipses in LAF format.
+
+    Shape:
+        - Input: :math:`(B, N, 5)`
+        - Output:  :math:`(B, N, 2, 3)`
+    Example:
+        >>> input = torch.ones(1, 10, 5)  # BxNx5
+        >>> output = kornia.ell2LAF(input)  #  BxNx2x3
+    """
+    n_dims = len(ells.size())
+    if (n_dims != 3):
+        raise TypeError(
+            "ellipse shape should be must be [BxNx5]. "
+            "Got {}".format(ells.size()))
+    B, N, dim = ells.size()
+    if (dim != 5):
+        raise TypeError(
+            "ellipse shape should be must be [BxNx5]. "
+            "Got {}".format(ells.size()))
+    ell_shape = torch.cat([torch.cat([ells[:, :, 2:3], ells[:, :, 3:4]], dim=2).unsqueeze(2),
+                           torch.cat([ells[:, :, 3:4], ells[:, :, 4:5]], dim=2).unsqueeze(2)],
+                           dim=2).view(-1, 2, 2)
+    out = torch.matrix_power(torch.cholesky(ell_shape, False), -1).view(B, N, 2, 2)
+    out = torch.cat([out, ells[:, :, :2].view(B, N, 2, 1)], dim=3)
+    return out
```

Comment:
typing and comment the code. Feel like some documented math should be included. A bit difficult to follow without knowing the exact maths